### PR TITLE
Remove most X509/globus links and hooks from PyCBC

### PR DESCRIPTION
**Pull request by spxiwh**
X509 and "grid certificates" are becoming increasingly unsupported and following OS upgrades on LVK clusters, now do not meet the requirements of search workflows within LVK.

As discussed in #4373 we are going to move over to scitokens.

scitokens are reasonably simple for the user to set up, so we will add scitokens support by instructing users to run the equivalent of `ligo-proxy-init`:

```
htgettoken -a vault.ligo.org -i igwn
```

to get tokens. Then the following line must be added into the `[pegasus_profile]` section of any workflow:

```
use_oauth_services = igwn
```

I'm not replacing the sanity checks in `pycbc_submit_dax` or the `start` script, because the former is deprecated, and the latter is only used on reruns (we never actually check X509 certificates explicitly in pycbc.workflow right now). If someone wanted to do that, I'd be happy to review.

### CODE REVIEW

File: docs/workflow/pycbc_make_coinc_search_workflow.rst
Comment (GarethCabournDavies) [inline docs/workflow/pycbc_make_coinc_search_workflow.rst:648]:
```suggestion
``https://`` URIs, if the data is not available locally.
```

File: docs/workflow/pycbc_make_coinc_search_workflow.rst
Comment (GarethCabournDavies) [inline docs/workflow/pycbc_make_coinc_search_workflow.rst:694]: Examples _may_ be good to retain?

File: pycbc/workflow/datafind.py
Comment (GarethCabournDavies) [inline pycbc/workflow/datafind.py:756]: In other places, the xrootd has been retained (e.g. the changes in `pycbc/workflow/pegasus_files/pegasus-properties.conf`) - should this be the same?

**Review by GarethCabournDavies — COMMENTED** [review]
This looks okay to me (without particular expertise), just a couple of minor comments. Will there be another PR for adding in scitoken functionality, or will they "just work"?

File: docs/workflow/pycbc_make_coinc_search_workflow.rst
Comment (spxiwh) [reply docs/workflow/pycbc_make_coinc_search_workflow.rst:694]: I don't know how we are supposed to do this anymore with scitokens. Perhaps we would just use `scp` or `ftp`?? I'll check the relevant syntax.

File: pycbc/workflow/datafind.py
Comment (spxiwh) [reply pycbc/workflow/datafind.py:756]: This one should go, as the OSG site is a LVK thing and this does not use root. I'm never sure what this was added for, but if non-LVK people still use this, it would be needed in pegasus-properties.conf (and removing that would be out of scope for this PR).

**Comment by spxiwh** [comment]
sciserver functionality would be added through config files ... I think that's easiest. Documentation for that will exist within LVK, does it need to be here as well?

**Comment by spxiwh** [comment]
It looks like we're still using a deprecated way of calling into gwdatafind, which does not support scitokens .... Will need to migrate that as well.

**Review by GarethCabournDavies — CHANGES_REQUESTED** [review]
The CI failures are wonderfully uninformative: `Internal error!`

I think after those are fixed this is good to go

**Comment by spxiwh** [comment]
Thanks @GarethCabournDavies I will fix that before proceeding, but I do need https://github.com/gwastro/pycbc/pull/4383 merged before this one, as datafind only supports scitokens if we move to the not-deprecated API.

**Comment by spxiwh** [comment]
This is now currently on hold while I wait for resolutions to two issues on the IGWN end. https://git.ligo.org/computing/helpdesk/-/issues/4079 and https://git.ligo.org/computing/helpdesk/-/issues/4088 prevent us from proceeding with the switch to scitokens for now. (@duncanmmacleod Is there anyone else I should poke on the IGWN end, as I am concerned about delays to O4 offline code readiness).

I did figure out why the tests are failing, and have a fix for that waiting to be rebased once the gwdatafind interface changes are in.

**Comment by spxiwh** [comment]
Okay, I'm just about ready to proceed with this. I thought I had understood why this was failing, but now cannot remember, so let's see if it's still failing after rebase to master. I would like to merge this after discussing at the next PyCBC LVK call.

**Review by GarethCabournDavies — APPROVED** [review]

**Pull request merged**
